src/app/hooks/useLocalStorage.ts

```diff
@@ -0,0 +1,17 @@
+import { useState } from 'react';
+
+export default function useLocalStorage<T>(
+  key: string,
+  initialValue: T
+): [storedValue: T, setValue: (value: T) => void] {
+  const [storedValue, setStoredValue] = useState<T>(() => {
+    const item = localStorage.getItem(key);
+    return item ? JSON.parse(item) : initialValue;
```

Comment:
This hook should do some error handling to make it less problematic if there is something in `localStorage` that shouldn't be there. You could ask Paul for example he knows more